### PR TITLE
[PyOV] Drop Python 3.6 support

### PR DESCRIPTION
**Pull request by p-wysocki**
### Details:
 - Remove Python 3.6 from docs, use Python 3.7 on GitHub workflows
 - Not to be merged until explicit permission given (see ticket)

### Tickets:
 - 86687

Notes:
- Example given in [tools/pot/docs/E2eExample.md](https://github.com/openvinotoolkit/openvino/blob/master/tools/pot/docs/E2eExample.md#end-to-end-command-line-interface-example-pot_configs_examples_readme) has been tested with Python 3.6, which will no longer be supported, but I cannot change it to 3.7.
- Supported OSes in [src/bindings/c/docs/api_overview.md](https://github.com/openvinotoolkit/openvino/blob/master/src/bindings/c/docs/api_overview.md#supported-oses) are out of date and need a separate update.

### CODE REVIEW

File: docs/MO_DG/prepare_model/Model_Optimizer_FAQ.md [hunk 1]
```diff
@@ -620,13 +620,13 @@ It means that you are trying to convert a topology contains the `_contrib_box_nm
...
+If a `*.caffemodel` file exists and is correct, the error occurred possibly because of the use of Python protobuf implementation. In some cases, error messages may appear during model parsing, for example: "`utf-8` codec can't decode byte 0xe0 in position 4: invalid continuation byte in field: mo_caffe.SpatialTransformerParameter.transform_type". You can either use Python 3.7 or build the `cpp` implementation of `protobuf` yourself for your version of Python. For the complete instructions about building `protobuf` from sources, see the appropriate section in the [Converting Models with Model Optimizer](../Deep_Learning_Model_Optimizer_DevGuide.md) guide.
```
Comment (rkazants) [inline docs/MO_DG/prepare_model/Model_Optimizer_FAQ.md:623]: Did we already mention about Python 3.6 support drop in the previous release?
I think we need to make a note and notify users about it and only after this is allowed to remove it overall.

Comment (brmarkus) [reply docs/MO_DG/prepare_model/Model_Optimizer_FAQ.md:623]: EXACTLY!
In the release-notes "https://www.intel.com/content/www/us/en/developer/articles/release-notes/openvino-relnotes-2021.html" Ubuntu 18.04 is moved to "supported with limitations", but nothing said about deprecation of Python 3.6.
Also the latest release-notes don't announce neither the deprecation nor removal of Python v3.6.
As long as Ubuntu 18.04LTS is supported Python v3.6 "still exists".

Comment (mlukasze) [reply docs/MO_DG/prepare_model/Model_Optimizer_FAQ.md:623]: Good comments, thank you for this!
We will discuss it internally on one of next meetings.
Decision and consequences to stay with support of py3.6 is a little bit more complicated, so we are taking each point of view to our considerations carefully.

File: .github/github_org_control/configs.py [hunk 2]
```diff
@@ -14,8 +14,8 @@
 from pathlib import Path
 
 
-if sys.hexversion < 0x3060000:
-    raise Exception("Python version must be >= 3.6")
+if sys.version_info.minor < 7:
```
Comment (artkuli) [inline .github/github_org_control/configs.py:17]: IMO when we check version of the package, we need to check major and minor number of version

File: src/bindings/python/setup.cfg [hunk 3]
```diff
@@ -78,8 +78,8 @@ show_column_numbers = True
 show_error_context = True
 show_absolute_path = True
 pretty = True
-follow_imports=normal
+follow_imports = normal
 disallow_untyped_defs = True
-disallow_untyped_calls = True
+disallow_untyped_calls = False
```
Comment (artkuli) [inline src/bindings/python/setup.cfg:83]: Why did you change it?

File: .github/github_org_control/configs.py (hunk 2 above)
Comment (praasz) [reply .github/github_org_control/configs.py:17]: > IMO when we check version of the package, we need to check major and minor number of version

Consider check version:
```python
sys.version_info[:3] < (3, 6, 0)
```

Comment (p-wysocki) [reply .github/github_org_control/configs.py]: Thank you for comments, I applied @praasz suggestion, really cool way of checking Python version. :)

File: src/bindings/python/setup.cfg (hunk 3 above)
Comment (p-wysocki) [reply src/bindings/python/setup.cfg]: Due to raising Python version in GitHub workflows a higher version of `numpy` was installed, which caused `mypy` to have issues with it's untyped functions. It is one of the symptoms of recent weird flake8/mypy behaviour I talked about earlier - today the issues do not reproduce locally. I changed it back to `True`, but `mypy` finds errors in `numpy` functions:
`Call to untyped function "np.concatenate" in typed context`
It seems like the function is not typed, but it's a third party library. I'll see if I can solve it in a more elegant way than changing the config.

File: tools/deployment_manager/configs/linux.json [hunk 4]
```diff
@@ -87,14 +87,6 @@
...
       ]
     },
-    "python3.6": {
-      "ui_name": "OpenVINO Python API for Python3.6",
-      "group": ["python"],
-      "dependencies" : ["ie_core"],
-      "files": [
-        "python/python3.6"
-      ]
-    },
```
Comment (ilya-lavrenov) [inline tools/deployment_manager/configs/linux.json:97]: In general, I want to keep support of 3.6 version for open source users. Can we revert all places in code which prohibit to do it?

Comment (p-wysocki) [reply tools/deployment_manager/configs/linux.json]: Done, I reverted all asserts for Python version to be 3.7 back to 3.6, open source users will not encounter an assertion error when using 3.6.

File: src/bindings/python/setup.cfg (hunk 3 above)
Comment (p-wysocki) [reply src/bindings/python/setup.cfg]: I ignored type checking in 4 lines which were using problematic `numpy` functions instead of disabling the check altogether.

**Review by ilya-lavrenov — APPROVED** [review]

**Review by praasz — APPROVED** [review]

**Review by artkuli — APPROVED** [review]

**Review by ilyachur — APPROVED** [review]

File: .github/github_org_control/configs.py [hunk 5]
```diff
@@ -14,7 +14,7 @@
 from pathlib import Path
 
 
-if sys.hexversion < 0x3060000:
+if sys.version_info[:2] < (3, 6):
     raise Exception("Python version must be >= 3.6")
```
Comment (jiwaszki) [inline .github/github_org_control/configs.py:18]: Let's clarify this and change if necessary.

**Review by jiwaszki — APPROVED** [review]

File: src/bindings/python/docs/contributing.md [hunk 6]
```diff
@@ -4,6 +4,11 @@
 *To be added...*
 
 ##### Enviroment
+In case the Python version you have is not supported by OpenVINO, you can refer to 
+
+    openvino/src/bindings/python/docs/python_version_upgrade.md
```
Comment (ilya-lavrenov) [inline src/bindings/python/docs/contributing.md:9]: should we refer directly to github to make it to be a reference?

Comment (p-wysocki) [reply src/bindings/python/docs/contributing.md]: Good point for user convenience, done.
